test/AdaptiveSpinLockTest.cpp

```diff
@@ -62,27 +62,39 @@ TEST(AdaptiveSpinLockTest, WorksWithStdUniqueLock) {
   lock.unlock();
 }
 
-const int thread_num = 7;
-int val;
-pipert::AdaptiveSpinLock* g_lock = nullptr;
+namespace {
 
-void SharedIncrementor(int times) {
-  ASSERT_TRUE(g_lock);
-  for (int i = 0; i < times; i++) {
-    std::lock_guard<pipert::AdaptiveSpinLock> guard(*g_lock);
-    val++;
+class SharedIncrementStore {
+ public:
+  SharedIncrementStore(const int threads) : threads_(threads){};
+
+ public:
+  void Increment(int cycles) {
+    std::lock_guard<pipert::AdaptiveSpinLock> guard(slock_);
+    for (int i = 0; i < cycles; ++i) val_++;
   }
+  int GetVal() const { return val_; }
+
+ private:
+  pipert::AdaptiveSpinLock slock_;
+  const int threads_;
+  int val_;
+};
+
+static SharedIncrementStore& SharedIncrementorInstance(const int cycles) {
+  static SharedIncrementStore si(cycles);
+  return si;
 }
 
-TEST(AdaptiveSpinLockTest, ThreadsSyncedCorrectly) {
-  const int cycles = 100;
+}  // namespace
+
+TEST(AdaptiveSpinLockTest, ThreadsSyncCorrectly) {
+  const int cycles = 50000;
```

Comment:
Isn't it too much? How long will it run if you do individual locks? Maybe I'm wrong.